### ISSUE TITLE
enumconstruct.cs Code Smell

### ISSUE THREAD

**Issue by GoogleCodeExporter** [barsu8184/google-gdata#90]
```
Google.GData.Extensions.EnumConstruct contains a public method "equals",
starting on line 99 in version 1.1.1.0 and SVN as of 2007-11-01.

This method is summarized as "Equal operator overload".  However, this
isn't an operator overload, nor does it override the .NET built-in "Equals"
(case-sensitive) method of System.Object.

Furthermore, the code is missing the "override" keyword.  The
implementation suggests subclasses might override this implementation - but
since the implementation is neither marked "override" nor "virtual", no
subclass can.

Finally, the code isn't used anywhere (removing it doesn't affect
compilation).  

Suggest removal.
```

Original issue reported on code.google.com by `eamon.nerbonne` on 1 Nov 2007 at 11:31

**Comment by GoogleCodeExporter** [comment]
```
removing it would certainly not effect compilation, as the compiler would 
auto-generate a comparison 
operator. The intention, if i read the comments of the person who wrote code 
are to provide a way to compare 
2 simple enum based on object type and value. 

change the code to do the right thing, method is now declared as:

      public override bool Equals(Object o)




```

Original comment by `fman...@gmail.com` on 13 Nov 2007 at 2:18
- Changed state: **Started**

**Comment by GoogleCodeExporter** [comment]
```
I'm supposing you're using a custom EnumConstruct (as opposed to plain enum) so
unknown values (such as potentially new values) can be contained in the 
enumconstruct?
```

Original comment by `eamon.nerbonne` on 15 Nov 2007 at 10:54

**Comment by GoogleCodeExporter** [comment]
```
Fixed / changed by release 1.1.2. 

and yes. That is one of the reasons for this class. It might be going away 
whenever
this code get's revisited, but until that time, it sticks for backwards compat. 
```

Original comment by `fman...@gmail.com` on 15 Nov 2007 at 5:55

**Comment by GoogleCodeExporter** [comment]
Original comment by `fman...@gmail.com` on 15 Nov 2007 at 5:55
- Changed state: **Fixed**
